public/tier1/mempool.h

```diff
@@ -127,19 +108,26 @@ class CMemoryPoolMT : public CUtlMemoryPool
 // and construction and destruction of objects.
 //-----------------------------------------------------------------------------
 template< class T >
-class CClassMemoryPool : public CUtlMemoryPool
+class CUtlMemoryPool : public CUtlMemoryPoolBase
 {
 public:
-	CClassMemoryPool(int numElements, int growMode = GROW_FAST, int nAlignment = 0 ) :
-		CUtlMemoryPool( sizeof(T), numElements, growMode, MEM_ALLOC_CLASSNAME(T), nAlignment ) {}
+	CUtlMemoryPool( int numElements, MemoryPoolGrowType_t growMode = UTLMEMORYPOOL_GROW_FAST, const char *pszAllocOwner = MEM_ALLOC_CLASSNAME(T), MemAllocAttribute_t allocAttribute = MemAllocAttribute_Unk0 ) 
+		: CUtlMemoryPoolBase( sizeof(T), numElements, alignof(T), growMode, pszAllocOwner, allocAttribute ) {}
 
 	T*		Alloc();
 	T*		AllocZero();
 	void	Free( T *pMem );
-
 	void	Clear();
 };
 
+
+//-----------------------------------------------------------------------------
+// Multi-thread/Thread Safe Memory Class
+//-----------------------------------------------------------------------------
+template< class T >
+using CUtlMemoryPoolMT = CUtlMemoryPool<T>;

```

Comment:
I assume this is a stub class? Some note that it's an incomplete/stubbed class would be nice in this case

Reply:
CUtlMemoryPoolMT is the same as CUtlMemoryPool.
It seems to be used to avoid changing names in places where CUtlMemoryPoolMT is used when moving to s2.
![Безымянный22222](https://github.com/alliedmodders/hl2sdk/assets/152704532/7aa665ed-7609-45a7-ae2c-78af15ebadc7)

Reply:
So did they just threw MT version of it to the bin?

Reply:
It seems so, since CUtlMemoryPoolBase itself is threadsafe.

Reply:
Ah, then alright